pkg/crc/machine/start.go

```diff
@@ -163,6 +163,10 @@ func (client *client) Start(ctx context.Context, startConfig types.StartConfig)
 			return nil, errors.Wrap(err, "Error getting bundle metadata")
 		}
 
+		if err := bundleMismatchWithPreset(startConfig.Preset, crcBundleMetadata); err != nil {
+			return nil, err
+		}
+
```

Comment:
Isn't this only triggered at VM creation time though? And now for example `crc start; crc stop; crc config set preset podman; crc start` will not return an error?

Reply:
@cfergeau In that scenario as soon as user do `crc config set preset podman`, we show the warning about `crc delete => crc setup => crc start` as callback. If user ignore this message then it should start the vm with same preset which user set on the fresh start. I think that is expected behavior?

Reply:
> If user ignore this message then it should start the vm with same preset which user set on the fresh start. I think that is expected behavior?

Well, this is a change of behaviour introduced by this commit which was not mentioned in the commit log, so I pointed it out. You are right that if the user does not run delete, reusing the same preset as what the VM was created with is the right thing to do. However, won't the preset get in the way, as they'll be done for the new preset rather than the previous ones?

Reply:
@cfergeau Agreed, I moved this bit very early on the start method so now it should handle both the cases.